pkgs/applications/graphics/ktikz/default.nix

```diff
@@ -1,70 +1,58 @@
-{ stdenv, fetchFromGitHub, gettext, poppler, qt5 , pkgconfig }:
-
-# Warning: You will also need a working pdflatex installation containing
-# at least auctex and pgf.
+{ stdenv, fetchFromGitHub, fetchpatch
+, pkgconfig, makeWrapper
+, poppler, qt5, gnuplot
+}:
 
 # This package only builds ktikz without KDE integration because KDE4 is
 # deprecated and upstream does not (yet ?) support KDE5.
 # See historical versions of this file for building ktikz with KDE4.
 
 stdenv.mkDerivation rec {
-  version = "unstable-20161122";
-  name = "qtikz-${version}";
-
-  src = fetchFromGitHub {
-    owner = "fhackenberger";
-    repo = "ktikz";
-    rev = "be66c8b1ff7e6b791b65af65e83c4926f307cf5a";
-    sha256 = "15jx53sjlnky4yg3ry1i1c29g28v1jbbvhbz66h7a49pfxa40fj3";
-  };
+  version = "0.12";
+  pname = "qtikz";
 
   meta = with stdenv.lib; {
     description = "Editor for the TikZ language";
+    homepage = "https://github.com/fhackenberger/ktikz";
     license = licenses.gpl2;
     platforms = platforms.linux;
     maintainers = [ maintainers.layus ];
+    longDescription = ''
+      You will also need a working *tex installation in your PATH, containing at least `preview` and `pgf`.
```

Comment:
Why not add them to path like gnuplot?

Reply:
Because you cannot install latex packages "alone", you need something alogn the lies of `texlive.withPackages`. We could provide a minimal working environment, but that would be mostly redundant with the user-defined one. Or maybe not ?

Reply:
I don't fully understand how texlive works in nixos but it is my understanding that adding a minimal texlive to this package wouldn't interfere much / increase store size  if the user had a bigger/different texlive installed. If that is the case it would surely be worth including this minimal texlive and using makeWrapper to include it in the path. Apologies if my rambling doesn't make a lot of sense

Reply:
This remark makes total sense. My concern is that providing a minimal environment will make qtikz break later on, in unexpected ways. I think it is better to provide none, so that users get a proper error on the first use, and know that they have to provide a working TeX environment.

Providing a default environment would help, but then users will start wondering how to add packages to it. They may also have issues to know if they are using their own environment instead of the default one, etc.

But it kind of boils down to personal tastes. My use of qtikz (and from the time it has been broken, it seems I am the only user) is to generate figures that will be included in larger, complete documents. These figures use packages from that document, and extra tikz libraries from my tex install. A default tex installation is nearly useless for me and would, as stated above, be more confusing that helpful.

This question has a broader scope than this lone package. I guess it arises in other applications. I should maybe start a ML thread...

Reply:
I think I agree with you. I don't understand why this would cause the package to break but I had more of a think and I can see the situation where someone would like to use qtikz and write something that requires extra tex packages (say for the labels). In this case it makes most sense that qtikz uses whatever tex environment the user wants to use. Also I assume anyone using this program will have tex installed

Reply:
I feel the ideal is to somehow use a "union" of a minimal tex needed for this to work and the current environment tex though I don't think this is possible

Reply:
> I feel the ideal is to somehow use a "union" of a minimal tex needed for this to work and the current environment tex though I don't think this is possible

Yes, that would be the best, but I know of no way to do that.

> I don't understand why this would cause the package to break

Well, maybe I am over-cautious here.